R/get_types_recipes.R

```diff
@@ -0,0 +1,112 @@
+## Buckets variables into discrete, mutally exclusive types
+get_types <- function(x) {
+  res <- lapply(x, get_types_recipes)
+  tibble(variable = names(res), type = unname(res))
+}
+
+#' Get types for use in recipes
+#'
+#' The `get_types_recipes()` generic is used internally to supply types to
+#' columns used in recipes. These functions underlie the work that the user sees
+#' in [selections].
+#'
+#' This function acts a extended recipes-specific version of [class()]. By
+#' ignoring differences in similar types ("double" and "numeric") and allowing
+#' each element to have multiple types ("factor" returns "factor", "unordered",
+#' and "nominal". And "character" returns "string", "unordered", and
+#' "nominal") we are able to create more natural selectors such as
+#' [all_nominal()], [all_string()] and [all_integer()].
+#'
+#' @export
+#' @param x A character vector
+#' @examplesIf rlang::is_installed("modeldata")
+#'
+#' data(Sacramento, package = "modeldata")
+#' lapply(Sacramento, get_types_recipes)
+get_types_recipes <- function(x) {
```

Comment:
This S3 generic is made such that any extension package can define types for other classes without recipes having to be modified

Reply:
I agree. `tune` has a lot of `.get_*()` functions and these are helpful in downstream packages like `finetune`.

I thing that, since`get_types()` works on tibbles, something like `.get_tibble_data_types()` is a good idea. Maybe call the generic `.get_data_types()`?